### PR TITLE
Fix issue with :skip: introduced by :include: feature

### LINKED ISSUE #141: :skip: no longer works with standard library objects
Prior to the 0.14 release, it was possible to use `:skip:` on standard library objects that were imported by a module:

```
.. automodapi:: sphinx_automodapi.tests.example_module.stdlib
    :skip: time
    :skip: Path
```

In 0.14, those `:skip:` directives are disappeared by code introduced to support `:include:`.  I opened a PR with a test showing the new failure: https://github.com/astropy/sphinx-automodapi/pull/140.  The test passes if I revert the commits from #127.

### PR DESCRIPTION
**Pull request by adrn**
This is a fix for the bug identified in #141. The fix is implemented in the first commit from me, which simplifies some of the logic and reverts back to looking more like it did before #127 but keeps the new `:include:` functionality.

Note: this includes the commits from #140 which added a useful regression test.

cc @eslavich

EDIT:

* Fix #141 
* Close #140

### CODE REVIEW

**Comment by adrn** [comment]
Uhh, what's up with the failing tests?? Any ideas @pllim?

**Comment by pllim** [comment]
Looks like Sphinx is using distutils but it is deprecated now. I guess we can ignore it...

```
../../.tox/py38-test-sphinx35/lib/python3.8/site-packages/sphinx/util/smartypants.py:33: in <module>
    from sphinx.util.docutils import __version_info__ as docutils_version
../../.tox/py38-test-sphinx35/lib/python3.8/site-packages/sphinx/util/docutils.py:45: in <module>
    __version_info__ = tuple(LooseVersion(docutils.__version__).version)
../../.tox/py38-test-sphinx35/lib/python3.8/site-packages/setuptools/_distutils/version.py:53: in __init__
    warnings.warn(
E   DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
```

**Comment by pllim** [comment]
I don't grok what is going on here, so I'll wait for @eslavich to review. Thanks for the quick fix!

File: sphinx_automodapi/automodapi.py
Comment (eslavich) [inline sphinx_automodapi/automodapi.py:415]: I think the new `:include:` feature has a similar problem as was introduced for `:skip:`.  Here's a test that fails on this branch:

```python
am_replacer_include_stdlib_str = """
This comes before

.. automodapi:: sphinx_automodapi.tests.example_module.stdlib
    :include: add

This comes after
"""

am_replacer_include_stdlib_expected = """
This comes before


sphinx_automodapi.tests.example_module.stdlib Module
----------------------------------------------------

.. automodule:: sphinx_automodapi.tests.example_module.stdlib

Functions
^^^^^^^^^

.. automodsumm:: sphinx_automodapi.tests.example_module.stdlib
    :functions-only:
    :toctree: api
    :skip: Path,time


This comes after
""".format(empty='')


def test_am_replacer_include_stdlib(tmpdir):
    """
    Tests using the ":include: option in an ".. automodapi::"
    in the presence of objects imported from the standard library.
    """

    with open(tmpdir.join('index.rst').strpath, 'w') as f:
        f.write(am_replacer_include_stdlib_str.format(options=''))

    run_sphinx_in_tmpdir(tmpdir)

    with open(tmpdir.join('index.rst.automodapi').strpath) as f:
        result = f.read()

    assert result == am_replacer_include_stdlib_expected
    ```

**Review by eslavich — COMMENTED** [review]
One possible issue, thanks for the fast turnaround on this @adrn!

File: sphinx_automodapi/automodapi.py
Comment (adrn) [reply sphinx_automodapi/automodapi.py:415]: I don't think this is an issue -- you just have to add the stdlib names as `:allowed-package-names:`. For example,

```python
am_replacer_include_stdlib_str = """
This comes before

.. automodapi:: sphinx_automodapi.tests.example_module.stdlib
    :include: add
    :allowed-package-names: pathlib, datetime, sphinx_automodapi

This comes after
"""

am_replacer_include_stdlib_expected = """
This comes before


sphinx_automodapi.tests.example_module.stdlib Module
----------------------------------------------------

.. automodule:: sphinx_automodapi.tests.example_module.stdlib

Functions
^^^^^^^^^

.. automodsumm:: sphinx_automodapi.tests.example_module.stdlib
    :functions-only:
    :toctree: api
    :skip: Path,time
    :allowed-package-names: pathlib,datetime,sphinx_automodapi


This comes after
""".format(empty='')

Then the test passes.
```

Comment (eslavich) [reply sphinx_automodapi/automodapi.py:415]: Ah, okay!  That makes sense.

**Review by eslavich — APPROVED** [review]
LGTM

File: sphinx_automodapi/automodapi.py
Comment (pllim) [reply sphinx_automodapi/automodapi.py:415]: Do you at least want to put this test in CI before merge?

Comment (adrn) [reply sphinx_automodapi/automodapi.py:415]: Sure

Comment (eslavich) [reply sphinx_automodapi/automodapi.py:415]: I'm not sure it's a worthy test, given what @adrn told me -- it makes sense that `:skip:` doesn't appear, but I'm not sure that we need to assert that it is not present.

Comment (adrn) [reply sphinx_automodapi/automodapi.py:415]: I added one more check that doing what I said in the comment above makes your test pass -- I think that is still worth including!

**Review by pllim — APPROVED** [review]
Thanks, all!

**Comment by pllim** [comment]
Since I also fixed `jwst` docs to not require this patch, I don't think an immediate release is needed. But please let me know if you disagree. Thanks!

**Comment by mwcraig** [comment]
@pllim -- if you have a chance a new release would be helpful because `ccdproc` tests are failing because of this (we have defined `__all__` so that fix won't work here).

Failing test: https://github.com/astropy/ccdproc/runs/4757634889?check_suite_focus=true#step:7:79

Pinned the version for now, so no an emergency

**Comment by pllim** [comment]
Done. v0.14.1 is on PyPI. Hope this helps!

**Pull request merged**
